pkg/encrypt/encrypt_test.go

```diff
@@ -64,6 +75,6 @@ func (t *testEncryptSuite) TestEncrypt(c *C) {
 	c.Assert(err, NotNil)
 
 	// invalid content
-	_, err = Decrypt(ciphertext[1:])
+	_, err = Decrypt(removeChar(ciphertext, ivSep[0]))
```

Comment:
are there something wrong with our implemention? I think old case is right

Reply:
ping @amyangfei

Reply:
https://github.com/pingcap/dm/blob/39e09ddcf97bb6752ecdfba4e0c7c96826bc783b/pkg/encrypt/encrypt.go#L77-L79
the decrypt function checks the specific byte is `iv separator` only. In the old case, the ciphertext is  
```
[iv of block size, often 16] + @ + encrypted-plaintext
```
and we remove the first byte, the ciphertext becomes
```
[the last 15 bytes of original iv + @] + `iv separator` + [new encrypted-plaintext]
```
We can always decrypt the new encrypted-plaintext with new iv, as long long some length condition is satisfied. Our encrypt mechanism doesn't maintain enough information to decide whether the new ciphertext is valid

Reply:
I know it, we should add a special case for this `fail` case